sway-parse/src/token.rs

```diff
@@ -130,7 +130,19 @@ pub fn lex_commented(
         if character == '/' {
             match l.stream.peek() {
                 Some((_, '/')) => {
-                    token_trees.push(lex_line_comment(&mut l, end, index));
+                    let has_newline = src[..index]
```

Comment:
Hmm, the thought behind this is that `take_while()` is only doing work while the predicate is true (`c.is_whitespace()`), and counting the newlines within that iteration, so its likely that the iterator isn't checking the entire `src` until the start (start since i call `rev()`)

Reply:
I mean that `src[..index]` will start at the very beginning of all of the `src` for the module, no? The provided `..index` range means "everything up until this index" - is that actually what's intended?

Reply:
Hm, at the time of writing it was the quick and dirty way (to me) to get a new reversed iterator to start looking backwards for newline tokens.

I'm not sure if this has cost implications, if this is what the question is about, but in terms of correctness it should stop at the first non-whitespace character regardless within the source code.

I added a new `search_end` in my latest commit to end the search at the last token's `span().end()` instead, but I'm not sure if this overcomplicates the problem.

Reply:
I marked this as resolved for now so I could get this merged since it changes some of the key code that causes conflicts - we could address this in a followup PR if it's an issue! @mitchmindtree